### PR TITLE
First Init.

### PR DESCRIPTION
**Pull request by Astoneti**
First  request.

### CODE REVIEW

File: Godel.Microservice.CurrencyRates/Business/Dto/CurrencyDto.cs [hunk 1]
```diff
@@ -0,0 +1,11 @@
+﻿namespace Godel.Microservice.CurrencyRates.Business.Dto
+{
+    public class CurrencyDto
+    {
+        public int Id { get; set; }
+
+        public string CurrencyName { get; set; }
```
Comment (rodchenkov) [inline Godel.Microservice.CurrencyRates/Business/Dto/CurrencyDto.cs]: please rename without Currency as class has this already

File: Godel.Microservice.CurrencyRates/Business/Dto/CurrencyDto.cs [hunk 2]
```diff
@@ -0,0 +1,11 @@
...
+
+        public string CurrencyCode { get; set; }
```
Comment (rodchenkov) [inline Godel.Microservice.CurrencyRates/Business/Dto/CurrencyDto.cs]: please rename without Currency as class has this already

File: Godel.Microservice.CurrencyRates/Business/Services/CurrencyService.cs [hunk 3]
```diff
@@ -0,0 +1,27 @@
+﻿using AutoMapper;
+using Godel.Microservice.CurrencyRates.Business.Contracts;
+using Godel.Microservice.CurrencyRates.Business.Dto;
+using Godel.Microservice.CurrencyRates.Data.Contracts;
+using Godel.Microservice.CurrencyRates.Data.Entities;
+using System.Collections.Generic;
+
+namespace Godel.Microservice.CurrencyRates.Business.Services
+{
+    public class CurrencyService : ICurrencyService
```
Comment (rodchenkov) [inline Godel.Microservice.CurrencyRates/Business/Services/CurrencyService.cs]: please fix order of items in class

File: Godel.Microservice.CurrencyRates/Controllers/CurrencyController.cs [hunk 4]
```diff
@@ -0,0 +1,33 @@
...
+    public class CurrencyController : ControllerBase
+    {
+        public CurrencyController(ILogger<CurrencyController> logger, ICurrencyService currencyService, IMapper mapper)
+        {
+            _logger = logger;
+            _currencyService = currencyService;
+            _mapper = mapper;
+        }
+
+        private readonly ILogger<CurrencyController> _logger; //Will be needed in further development!
```
Comment (rodchenkov) [inline Godel.Microservice.CurrencyRates/Controllers/CurrencyController.cs]: please remove unused code

File: Godel.Microservice.CurrencyRates/Controllers/CurrencyController.cs [hunk 5]
```diff
@@ -0,0 +1,33 @@
...
+using Godel.Microservice.CurrencyRates.Model;
+using Microsoft.AspNetCore.Mvc;
+using Microsoft.Extensions.Logging;
+using System.Collections.Generic;
+
+namespace Godel.Microservice.CurrencyRates.Controllers
+{
+    [ApiController]
+    [Route("[controller]")]
+    public class CurrencyController : ControllerBase
```
Comment (rodchenkov) [inline Godel.Microservice.CurrencyRates/Controllers/CurrencyController.cs:13]: please fix order of items in class

File: Godel.Microservice.CurrencyRates/Business/Services/CurrencyService.cs [hunk 6]
```diff
@@ -0,0 +1,27 @@
+﻿using AutoMapper;
+using Godel.Microservice.CurrencyRates.Business.Contracts;
+using Godel.Microservice.CurrencyRates.Business.Dto;
+using Godel.Microservice.CurrencyRates.Data.Contracts;
+using Godel.Microservice.CurrencyRates.Data.Entities;
+using System.Collections.Generic;
+
+namespace Godel.Microservice.CurrencyRates.Business.Services
```
Comment (rodchenkov) [inline Godel.Microservice.CurrencyRates/Business/Services/CurrencyService.cs]: move to upper folder as it is main point in business

File: Godel.Microservice.CurrencyRates/Data/CurrencyDbContext.cs [hunk 7]
```diff
@@ -0,0 +1,15 @@
+﻿using Godel.Microservice.CurrencyRates.Data.Entities;
+using Microsoft.EntityFrameworkCore;
+
+namespace Godel.Microservice.CurrencyRates.Data
+{
+    public class CurrencyDbContext : DbContext
+    {
+        public DbSet<CurrencyEntity> Currencies { get; set; }
```
Comment (rodchenkov) [inline Godel.Microservice.CurrencyRates/Data/CurrencyDbContext.cs:8]: please change order of items in class

File: Godel.Microservice.CurrencyRates/Data/Entities/CurrencyEntity.cs [hunk 8]
```diff
@@ -0,0 +1,11 @@
+﻿namespace Godel.Microservice.CurrencyRates.Data.Entities
+{
+    public class CurrencyEntity
+    {
+        public int Id { get; set; }
+
+        public string CurrencyName { get; set; }
```
Comment (rodchenkov) [inline Godel.Microservice.CurrencyRates/Data/Entities/CurrencyEntity.cs]: please remove Currency as it is in class name

File: Godel.Microservice.CurrencyRates/Data/Entities/CurrencyEntity.cs [hunk 9]
```diff
@@ -0,0 +1,11 @@
...
+
+        public string CurrencyCode { get; set; }
```
Comment (rodchenkov) [inline Godel.Microservice.CurrencyRates/Data/Entities/CurrencyEntity.cs]: please remove Currency as it is in class name

File: Godel.Microservice.CurrencyRates/Data/Repositories/CurrencyRepository.cs [hunk 10]
```diff
@@ -0,0 +1,12 @@
+﻿using Godel.Microservice.CurrencyRates.Data.Contracts;
+using Godel.Microservice.CurrencyRates.Data.Entities;
+
+namespace Godel.Microservice.CurrencyRates.Data.Repositories
```
Comment (rodchenkov) [inline Godel.Microservice.CurrencyRates/Data/Repositories/CurrencyRepository.cs:4]: please move to upper folder level

File: Godel.Microservice.CurrencyRates/Data/Repositories/CurrencyRepository.cs [hunk 11]
```diff
@@ -0,0 +1,12 @@
...
+{
+    public class CurrencyRepository : BaseRepository<CurrencyEntity>, ICurrencyRepository
+    {
+        public CurrencyRepository(CurrencyDbContext dbContext) : base(dbContext)
```
Comment (rodchenkov) [inline Godel.Microservice.CurrencyRates/Data/Repositories/CurrencyRepository.cs]: please move : base to new line to make it easier to see inheritance

File: Godel.Microservice.CurrencyRates/Godel.Microservice.CurrencyRates.sln [hunk 12]
```diff
@@ -0,0 +1,25 @@
+﻿
```
Comment (rodchenkov) [inline Godel.Microservice.CurrencyRates/Godel.Microservice.CurrencyRates.sln:1]: remove this solution

File: Godel.Microservice.CurrencyRates/MappingHelper/MappingProfile.cs [hunk 13]
```diff
@@ -0,0 +1,16 @@
+﻿using AutoMapper;
+using Godel.Microservice.CurrencyRates.Business.Dto;
+using Godel.Microservice.CurrencyRates.Data.Entities;
+using Godel.Microservice.CurrencyRates.Model;
+
+namespace Godel.Microservice.CurrencyRates.MappingHelper
```
Comment (rodchenkov) [inline Godel.Microservice.CurrencyRates/MappingHelper/MappingProfile.cs]: please change name from MappingHelper to Mappings

File: Godel.Microservice.CurrencyRates/MappingHelper/MappingProfile.cs [hunk 14]
```diff
@@ -0,0 +1,16 @@
...
+{
+    public class MappingProfile : Profile
```
Comment (rodchenkov) [inline Godel.Microservice.CurrencyRates/MappingHelper/MappingProfile.cs]: it is better to use per model profile and call it like CurrencyProfile, UserProfile etc

File: Godel.Microservice.CurrencyRates/Models/CurrencyModel.cs [hunk 15]
```diff
@@ -0,0 +1,11 @@
+﻿namespace Godel.Microservice.CurrencyRates.Model
+{
+    public class CurrencyModel
+    {
+        public int Id { get; set; }
+
+        public string CurrencyName { get; set; }
```
Comment (rodchenkov) [inline Godel.Microservice.CurrencyRates/Models/CurrencyModel.cs]: please remove Currency as it is already in class name

File: Godel.Microservice.CurrencyRates/Models/CurrencyModel.cs [hunk 16]
```diff
@@ -0,0 +1,11 @@
...
+
+        public string CurrencyCode { get; set; }
```
Comment (rodchenkov) [inline Godel.Microservice.CurrencyRates/Models/CurrencyModel.cs]: please remove Currency as it is already in class name

**Review by rodchenkov — CHANGES_REQUESTED** [review]
Please fix comments

File: Godel.Microservice.CurrencyRates/Business/Dto/CurrencyDto.cs (hunk 1 above)
Comment (Astoneti) [reply Godel.Microservice.CurrencyRates/Business/Dto/CurrencyDto.cs]: was renamed

File: Godel.Microservice.CurrencyRates/Business/Services/CurrencyService.cs (hunk 3 above)
Comment (Astoneti) [reply Godel.Microservice.CurrencyRates/Business/Services/CurrencyService.cs]: was fixed

File: Godel.Microservice.CurrencyRates/Controllers/CurrencyController.cs (hunk 4 above)
Comment (Astoneti) [reply Godel.Microservice.CurrencyRates/Controllers/CurrencyController.cs]: unused code was removed

File: Godel.Microservice.CurrencyRates/MappingHelper/MappingProfile.cs (hunk 14 above)
Comment (Astoneti) [reply Godel.Microservice.CurrencyRates/MappingHelper/MappingProfile.cs]: was renamed

File: Godel.Microservice.CurrencyRates/MappingHelper/MappingProfile.cs (hunk 13 above)
Comment (Astoneti) [reply Godel.Microservice.CurrencyRates/MappingHelper/MappingProfile.cs]: Was renamed from MappingHelper to Mappings

File: Godel.Microservice.CurrencyRates/Controllers/CurrencyController.cs (hunk 5 above)
Comment (Astoneti) [reply Godel.Microservice.CurrencyRates/Controllers/CurrencyController.cs:13]: was fixed

File: Godel.Microservice.CurrencyRates/Business/Dto/CurrencyDto.cs (hunk 2 above)
Comment (Astoneti) [reply Godel.Microservice.CurrencyRates/Business/Dto/CurrencyDto.cs]: was renamed

File: Godel.Microservice.CurrencyRates/Models/CurrencyModel.cs (hunk 15 above)
Comment (Astoneti) [reply Godel.Microservice.CurrencyRates/Models/CurrencyModel.cs]: was removed

File: Godel.Microservice.CurrencyRates/Models/CurrencyModel.cs (hunk 16 above)
Comment (Astoneti) [reply Godel.Microservice.CurrencyRates/Models/CurrencyModel.cs]: was removed

**Pull request closed without merging**